apps/authenticator/test/authenticator/sign_in/commands/resource_owner_test.exs

```diff
@@ -418,5 +549,46 @@ defmodule Authenticator.SignIn.Commands.ResourceOwnerTest do
 
       assert {:error, :unauthenticated} == Command.execute(input)
     end
+
+    test "fails if user totp do not match credential" do
+      scopes = RF.insert_list!(:scope, 3)
+      user = RF.insert!(:user)
+      totp = RF.insert!(:totp, user: user)
+      app = RF.insert!(:client_application)
+      hash = RF.gen_hashed_password("MyPassw@rd234")
+      password = RF.insert!(:password, user: user, password_hash: hash)
+
+      username = user.username
+      client_id = app.client_id
+      scope = scopes |> Enum.map(& &1.name) |> Enum.join(" ")
+
+      input = %{
+        username: username,
+        password: "MyPassw@rd234",
+        otp: "4321",
```

Comment:
Technically we have 1 / 1000 (0.1%) of chance of having a false positive here. I suggest fetching the right one and then choose a wrong value
Did I miss something?

Reply:
No because bytes default we generais cores with 6 digits.